.github/workflows/release.yml

```diff
@@ -1,90 +1,70 @@
-      - if: runner.os == 'Linux'
-        run: |
-          sudo apt-get update
-          sudo apt-get upgrade
-          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
-          sudo apt-add-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-10 main"
-          sudo apt-get update
-          sudo apt-get install -y libudev-dev libssl-dev libsasl2-dev libzstd-dev
-          sudo apt-get install -y openssl --allow-unauthenticated
-          sudo apt-get install -y libssl1.1 --allow-unauthenticated
-      - uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ env.RUST_VERSION }}
-          override: true
-          profile: minimal
-          components: rustfmt
-      - name: Build Plugin
-        run: |
-          echo "CI_TAG=${GITHUB_REF#refs/*/}" >> "$GITHUB_ENV"
-          echo "CI_OS_NAME=linux" >> "$GITHUB_ENV"
-          cargo build --release
-      - name: Build release tarball
-        run: ./ci/create-tarball.sh
-      - name: Publish to crates registry
-        run: |
-          cargo publish -p plerkle_serialization --token $CARGO_TOKEN --no-verify || true
-          sleep 30
-          cargo publish -p plerkle_messenger --token $CARGO_TOKEN --no-verify || true
-        shell: bash
-        env:
-          CARGO_TOKEN: ${{ secrets.CARGO_TOKEN }}
-      - name: Release
-        uses: softprops/action-gh-release@v1
-        if: startsWith(github.ref, 'refs/tags/')
-        with:
-          name: STABLE ${{ env.CI_TAG }} 
-          body: |
-            ## STABLE VERSION: 
-            Reccomended for Mainnet
-            ${{ env.GEYSER_PLUGIN_NAME }} ${{ env.CI_TAG }}
-            solana ${{ env.SOLANA_VERSION_STABLE }}
+      
+    - name: checkout repo
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+        lfs: true
+
+    - name: set env vars
+      run: |
+        source ci/env.sh
+        echo "RUST_VERSION=$rust_version" | tee -a "$GITHUB_ENV"
+        echo "RUST_PROFILE=$rust_profile" | tee -a "$GITHUB_ENV"
+        echo "RUST_COMPONENTS=$rust_components" | tee -a "$GITHUB_ENV"
+        echo "SOLANA_VERSION=$solana_version" | tee -a "$GITHUB_ENV"
+        echo "PLUGIN_VERSION=$plugin_version" | tee -a "$GITHUB_ENV"
+        echo "RELEASE=$release" | tee -a "$GITHUB_ENV"
+
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ env.RUST_VERSION }}
+        override: true
+        profile: ${{ env.RUST_PROFILE }}
+        components: ${{ env.RUST_COMPONENTS }}
+
+    - name: Build plerkle plugin
+      run: cargo build --verbose --release
+
+    - name: Package libplerkle
+      working-directory: target/release
+      run: tar -cvjf plerkle-x86_64-unknown-linux-gnu.tar.bz2 libplerkle.so
+
+    - name: Package libplerkle22
+      if: matrix.os == 'ubuntu-22.04'
+      working-directory: target/release
+      run: tar -cvjf plerkle2222-x86_64-unknown-linux-gnu.tar.bz2 libplerkle.so
+
+    - name: Publish to release
```

Comment:
This probably will create 2 releases, one per OS. Perhaps this should move to a separate job that runs just once (without the "matrix" strategy) and uses the binaries built on the previous job.

Reply:
This should create a single release, with both binaries.
If it's clearer, or the wildcard doesn't resolve correctly, we can do 2 publish to release actions, with if statement, so each os version publish the specific artifact. This can publish to the same release or to two completely different ones.